contracts/FundManagement.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract FundManagement is Ownable {
+    struct Fund {
+        uint256 totalAmount;
+        address asset;
+        uint256 requestWithdrawTime;
+        uint256 amountWithdrawable;
+    }
+
+    uint256 public requestWindow = 2 days;
+    mapping(address => mapping(uint256 => Fund)) public funds;
+    mapping(address => uint256) public numberOfFunds;
+    mapping(address => uint256) public locked;
+
+    event CreateFund(address indexed manager, address asset, uint256 amount);
+    event RequestFunds(address indexed to, uint256 amount);
+    event Claim(address indexed to, uint256 amount);
+    event Clawback(address account, uint256 fundNumber);
+
+    constructor() {}
+
+    function requestFunds(uint256 fundNumber, uint256 amount) external {
+        Fund storage fund = funds[msg.sender][fundNumber];
+        require(
+            fund.amountWithdrawable + amount <= fund.totalAmount, 
```

Comment:
gas optimisation: store `fund.amountWithdrawable + amount` in its own variable since its referenced twice